Dwifft/Dwifft.swift

```diff
@@ -75,20 +75,47 @@ public enum SectionedDiffStep<Section, Value>: CustomDebugStringConvertible {
     }
 }
 
+// Need to be able to count a sequence without materializing as an array in order to keep matchingEndsInfo below as fase as possible
+private extension Sequence {
+    func count() -> Int {
+        var i = 0
+        for _ in self {
+            i += 1
+        }
+        return i
+    }
+}
+
 /// Namespace for the `diff` and `apply` functions.
 public enum Dwifft {
 
+    internal static func matchingEndsInfo<Value: Equatable>(_ lhs: [Value], _ rhs: [Value]) -> (Int, ArraySlice<Value>, ArraySlice<Value>) {
```

Comment:
Even though it's an internal function, can you please add a docstring to this method to help with future debugging etc?